frontend/src/metabase/query_builder/actions/query-updates.ts

```diff
@@ -0,0 +1,21 @@
+import type { Dispatch, GetState } from "metabase-types/store";
+
+import * as Lib from "metabase-lib";
+import type { Limit } from "metabase-lib/types";
+
+import { getQuestion } from "../selectors";
+import { updateQuestion } from "./core";
+
+export const setLimit =
+  (limit: Limit) => (dispatch: Dispatch, getState: GetState) => {
+    const state = getState();
+    const question = getQuestion(state);
+    if (!question) {
+      return;
+    }
+    const query = question._getMLv2Query();
```

Comment:
This feels like it should be its own selector, but that doesn't need to be now.

Reply:
Yeah, agree. I'd like to see how it looks with a couple more things ported before trying to generalize/abstract this